app/models/miq_queue.rb

```diff
@@ -623,6 +623,16 @@ def self.display_name(number = 1)
 
   private
 
+  # NOTE: this will intentionally lookup missing zones every time
+  def validate_zone_name
+    if zone && !self.class.valid_zone_names[zone]
+      found = self.class.valid_zone_names[zone] = Zone.in_my_region.exists?(:name => zone)
+      errors.add(:zone, N_("Unknown Zone")) unless found
+    end
+  end
+
+  cache_with_timeout(:valid_zone_names, 1) { Hash.new }
```

Comment:
Is this 1 a 1.minute or 1.second.  Thinking we should be more explicit:
```suggestion
  cache_with_timeout(:valid_zone_names, 1.minute) { Hash.new }
```